pkg/utils/policy/policy_test.go

```diff
@@ -0,0 +1,212 @@
+package policy
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	mockclient "github.com/aws/aws-application-networking-k8s/mocks/controller-runtime/client"
+	anv1alpha1 "github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1"
+	"github.com/aws/aws-application-networking-k8s/pkg/model/core"
+)
+
+func Test_checkGatewayOrRouteTargetRefExistsForPolicy(t *testing.T) {
+	testNamespace := gwv1beta1.Namespace("test-ns")
+
+	notFoundError := fmt.Errorf("k8s resource not found")
+	type args struct {
+		policy core.Policy
+	}
+	tests := []struct {
+		name                         string
+		args                         args
+		expectedK8sClientReturnedObj client.Object
+		want                         bool
```

Comment:
want what?

Reply:
If this is just checking if we want the expected object to not be nil, shouldnt we be safe to assume that if the expectedK8sClientReturnedObj is nil, then an error is returned?

Also, expectedK8sClientReturnedObj is a bit long, can we shorten this to expectedObject?

Reply:
```
got, err := checkGatewayOrRouteTargetRefExistsForPolicy(ctx, k8sClient, tt.args.policy)
assert.Equalf(t, tt.want, got, "checkGatewayOrRouteTargetRefExistsForPolicy(%v, %v, %v)", ctx, k8sClient, tt.args.policy)
```
`want` is expected return value and `got` is actual return value, it's generated by goland idea any better name?

Reply:
I see, it's comparing expected/actual values. If this is a common pattern, then that's fine, I was just not familiar with it